File: src/models/product.js
class Product {
  constructor() {
    this.id = null;
    this.name = null;
    this.barcode = false;
    this.measure = "pcs";
    this.stock = null;
    this.tax = 0;
    this.purchased_price_wo_tax = null;
    this.purchased_price = null;
    this.selling_price = null;
    this.expiration_date = null;
    this.date_created = null;
    this.date_modified = null;
  }
  fromData(data) {
    this.id = data.id || null;
    this.name = data.name || null;
    this.barcode = data.barcode || false;
    this.measure = data.measure || "pcs";
    this.stock = data.stock || null;
    this.tax = data.tax || 0;
    this.purchased_price_wo_tax = data.purchased_price_wo_tax || null;
    this.purchased_price = data.purchased_price || null;
    this.selling_price = data.selling_price || null;
    this.expiration_date = data.expiration_date?.substring(0, 10) || null;
    this.date_created = data.date_created || null;
    this.date_modified = data.date_modified || null;
  }
}

export default Product;

File: resources/website/api_models/product.py
from flask_restx import fields

from .. import swagger

product_model = swagger.model("Product", {
    "id": fields.Integer,
    "barcode": fields.Integer,
    "name": fields.String,
    "stock": fields.Float,
    "tax": fields.Integer,
    "purchased_price_wo_tax": fields.Float,
    "purchased_price": fields.Float,
    "selling_price": fields.Float,
    "measure": fields.String,
    "expiration_date": fields.String,
    "date_created": fields.String,
    "date_modified": fields.String,
})

product_create_model = swagger.model("ProductCreate", {
    "barcode": fields.Integer,
    "name": fields.String,
    "stock": fields.Float,
    "tax": fields.Integer,
    "purchased_price_wo_tax": fields.Float,
    "purchased_price": fields.Float,
    "selling_price": fields.Float,
    "measure": fields.String,
    "expiration_date": fields.String,
})

products_delete_model = swagger.model("ProductsDelete", {
    "products": fields.List(fields.Integer),
})